File: solution-java/013_roman_to_integer.java
/*
	013. Roman to Integer
	@name: li jin
	@date: Feb 23, 2016
	@link: https://leetcode.com/problems/roman-to-integer/
	@time: 8 ms
*/

public class Solution {

    public int romanToInt(String s) {
        int result = 0;
        int [] roman_map = new int[256];
        roman_map['I'] = 1;
				roman_map['V'] = 5;
				roman_map['X'] = 10;
				roman_map['L'] = 50;
				roman_map['C'] = 100;
				roman_map['D'] = 500;
				roman_map['M'] = 1000;
        for (int i = 0; i < s.length() - 1; i++) {
            if (roman_map[s.charAt(i)] < roman_map[s.charAt(i + 1)]) result -= roman_map[s.charAt(i)];
            else result += roman_map[s.charAt(i)];
        }
        if (s.length() > 0)
            result += roman_map[s.charAt(s.length() - 1)];
        return result;
    }

}

File: solution-python/013_roman_to_integer.py
"""
	013. Roman to Integer
	@name: li jin
	@date: Feb 23, 2016
	@link: https://leetcode.com/problems/roman-to-integer/
	@time: 168 ms
"""

class Solution(object):
    def romanToInt(self, s):
        roman = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        ans, n = 0, len(s)
        for i in xrange(n - 1):
            if roman[s[i + 1]] > roman[s[i]]:
                ans -= roman[s[i]]
            else:
                ans += roman[s[i]]
        if n:
            ans += roman[s[-1]]
        return ans
